### PR TITLE
[tests-only][full-ci]Add tests for create link using root endpoint

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds tests for creating share link of drives using root endpoint

## Related Issue

- https://github.com/owncloud/ocis/issues/8810

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 1]
```diff
@@ -1630,3 +1630,611 @@ Feature: Create a share link for a resource
...
+                "pattern": "^%base_url%/s/[a-zA-Z]{15}$"
+              }
+            }
+          }
+        }
+      }
+      """
+
+
+  Scenario: try to create an internal share link of a project-space drive with password
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1897]:
```suggestion
  Scenario: try to create an internal link share of a folder inside project-space with password
```

Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNg/linkShare.feature:1897]: share link sound good than link share to me

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 2]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
       }
       """
     And the public should be able to download file "textfile.txt" from the last link share with password "%public%" and the content should be "to share"
+
+  @issue-7879
+  Scenario Outline: try to create a share link of a Personal and Share drive using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1635]:
```suggestion
  Scenario Outline: try to create a link share of a Personal and Share drives using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 3]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+      | createOnly       | Shares   |
+      | blocksDownload   | Shares   |
+      | view             | Personal |
+      | edit             | Personal |
+      | upload           | Personal |
+      | createOnly       | Personal |
+      | blocksDownload   | Personal |
+
+
+  Scenario Outline: try to create an internal share link of a Personal and Share drive using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1687]:
```suggestion
  Scenario Outline: try to create an internal link share of a Personal and Share drives using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 4]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+        }
+      }
+      """
+    Examples:
+      | drive    |
+      | Personal |
+      | Shares   |
+
+
+  Scenario Outline: try to create an internal share link with password of a Personal and Share drive using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1730]:
```suggestion
  Scenario Outline: try to create an internal link share with password of a Personal and Share drives using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 5]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+        }
+      }
+      """
+    Examples:
+      | drive    |
+      | Personal |
+      | Shares   |
+
+  @issue-7879
+  Scenario Outline: create a share link of a project-space drive with different permission using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1774]:
```suggestion
  Scenario Outline: create a link share of a project-space drive with different permissions using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 6]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "projectSpace" with the default quota using the Graph API
+    When user "Alice" tries to create the following space share link using root endpoint of the Graph API:
+      | space           | projectSpace  |
+      | permissionsRole | internal      |
+      | password        | %public%      |
+    Then the HTTP status code should be "400"
+
+  @issue-7879
+  Scenario Outline: create a share link of a project-space drive with display name and expiry date using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1908]:
```suggestion
  Scenario Outline: create a link share of a project-space drive with display name and expiry date using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 7]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+      | upload           |
+      | createOnly       |
+      | blocksDownload   |
+
+  @issue-7879
+  Scenario Outline: try to create a share link of a project-space drive with a password that is listed in the Banned-Password-List using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1980]:
```suggestion
  Scenario Outline: try to create a link share of a project-space drive with a password that is listed in the Banned-Password-List using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 8]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+      | ownCloud        | upload           |
+      | 123             | createOnly       |
+      | password        | createOnly       |
+      | ownCloud        | createOnly       |
+      | 123             | blocksDownload   |
+      | password        | blocksDownload   |
+      | ownCloud        | blocksDownload   |
+
+  @env-config @issue-7879
+  Scenario Outline: create a share link of a project-space drive without password using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:2036]:
```suggestion
  Scenario Outline: create a link share of a project-space drive without password using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 9]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+      | upload           |
+      | createOnly       |
+      | blocksDownload   |
+
+  @issue-7879
+  Scenario Outline: create a share link of a project-space drive with display name using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:2104]:
```suggestion
  Scenario Outline: create a link share of a project-space drive with display name using root endpoint
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 10]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
...
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+      | upload           |
+      | createOnly       |
+      | blocksDownload   |
+
+  @issue-7879
+  Scenario Outline: create a share link of a project-space drive with expiry date using root endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:2171]:
```suggestion
  Scenario Outline: create a link share of a project-space drive with expiry date using root endpoint
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 11]
```diff
@@ -1242,4 +1242,21 @@ public function userUpdatesTheLastDriveShareWithTheFollowingUsingRootEndpointOfT
 			)
 		);
 	}
+
+	/**
+	 * @When /^user "([^"]*)" (?:tries to create|creates) the following space share link using root endpoint of the Graph API:$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:1247]:
```suggestion
	 * @When /^user "([^"]*)" (?:tries to create|creates) the following space link share using root endpoint of the Graph API:$/
```

**Review by saw-jan — COMMENTED** [review]
we should call:
- user share
- group share
- link share

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 12]
```diff
@@ -1630,3 +1630,610 @@ Feature: Create a share link for a resource
       }
       """
     And the public should be able to download file "textfile.txt" from the last link share with password "%public%" and the content should be "to share"
+
+  @issue-7879
+  Scenario Outline: try to create a share link of a Personal and Share drive using root endpoint
+    When user "Alice" tries to create the following space share link using root endpoint of the Graph API:
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:1636]:
```suggestion
    When user "Alice" tries to create the following space link share using the Graph API:
```

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**